Clarify human review (#1531)
mozilla/extension-workshop: src/content/documentation/publish/what-does-review-rejection-mean-to-users.md

## Review overview

Any extension you submit to [addons.mozilla.org](https://addons.mozilla.org) (AMO) is subject to two reviews. There is a machine validation of your extension as part of the submission flow, and a human review that can take place at any time, sometimes before publication.

> The machine validation tells you immediately if anything needs to be fixed to enable the publication of your extension. The human review can occur at any time. In some cases, human review can be necessary for publication. When this review starts, the reviewer may ask for clarification about your extension. The outcome of the review could be the rejection of your extension’s latest version, and the rejection of earlier unreviewed versions if they also contain issues.

Reviewer: I would remove the repetition, we just told the reader a human review can occur anytime. Also, can we clarify under what circumstances a human review before publication is necessary?
```suggestion
The machine validation tells you immediately if anything needs to be fixed to enable the publication of your extension. In some cases, a human review may be necessary for publication. When this review starts, the reviewer may ask for clarification about your extension. The outcome of the review could be the rejection of your extension’s latest version and the rejection of earlier unreviewed versions if they also contain issues.
```

Author: The repetition was there in the original text as well, so I kept it.
We want to keep the current level of detail in the documentation.

Reviewer: Indeed, but no reason why we cannot take this opportunity to make a small improvement :-)

Author: Let's do that in a followup.
